src/components/ui/Modal.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import RModal from 'react-modal';
+
+export type ModalProps = {
+  className?: string;
+  triggerContent: string | JSX.Element;
+  triggerClassName?: string;
+  showClose?: boolean;
+} & React.PropsWithChildren;
+
+export interface ModalRef {
+  openModal: () => void;
+  closeModal: () => void;
+}
+
+function Modal({
```

Comment:
new generic Modal component that just adds some common functionality we want for our Modal components like a X close button, expose the openModal/closeModal as ref, etc